src/metric-engine/src/engine/read.rs

```diff
@@ -104,6 +99,15 @@ impl MetricEngineInner {
         }
     }
 
+    /// Returns true if it's a physical region.
+    pub async fn is_physical_region(&self, region_id: RegionId) -> bool {
+        self.state
```

Comment:
Is it possible to use std's rwlock for "self.state"? To avoid the async propagates to the "role" function. cc @waynexia